src/transaction_history.rs

```diff
@@ -351,6 +378,85 @@ fn increment_tx_count(store: &mut dyn Storage) -> StdResult<u64> {
     Ok(id)
 }
 
+fn store_tx_with_decoys(
+    store: &mut dyn Storage,
+    tx: &StoredExtendedTx,
+    for_address: &Addr,
+    block: &cosmwasm_std::BlockInfo,
```

Comment:
Hmm, we can, my concern is that it can somehow help the attacker to distinguish between a decoy and the actual address (Maybe length of the data or someth like that)